projects/AtomicData/service.nix

```diff
@@ -0,0 +1,67 @@
+{ lib, config, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.atomic-server;
+  envFile = pkgs.writeText ".env" ''
+    ATOMIC_CONFIG_DIR=/var/lib/atomic-server
+    ATOMIC_DATA_DIR=/var/lib/atomic-server
+    XDG_CACHE_HOME=/var/cache/atomic-server
+    ${generators.toINIWithGlobalSection {} {globalSection = cfg.settings;}}
+  '';
+in
+{ 
+  options = {
+    services.atomic-server = {
+      enable = mkEnableOption "Enable Atomic Server";
+      settings = mkOption {
+        default = {};
+        description = ''
+          Atomic Server configuration. Refer to <https://docs.atomicdata.dev/atomicserver/installation#atomicserver-cli-options--env-vars>
+          for details on supported values.   
+          ATOMIC_CONFIG_DIR and ATOMIC_DATA_DIR are set automatically to work with NixOS Modules.
+        '';
+         example = literalExpression ''
+          {
+            "ATOMIC_INITALIZE" = "true";
+            "ATOMIC_DOMAIN" = "localhost";
+            "ATOMIC_REBUILD_INDEX" = "false";
+            "ATOMIC_PORT" = "9883";
+          }
+        '';
+      };
+    };
+  };
+  config = mkIf cfg.enable {
+    users.users.atomic-server = {
+      isSystemUser = true;
+      group = "atomic-server";
+    };
+    users.groups.atomic-server = {};
+    systemd.services.atomic-server = {
+      description = "Atomic Server";
+      after = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+      serviceConfig = {
+        ExecStart = "${pkgs.atomic-server}/bin/atomic-server";
+        User = "atomic-server";
+        EnvironmentFile = envFile;
+        StateDirectory = "atomic-server";
+        CacheDirectory = "atomic-server";
```

Comment:
So IIRC we discovered that this [systemd option sets the `CACHE_DIRECTORY` env var](https://www.freedesktop.org/software/systemd/man/latest/systemd.exec.html), which turns out to ignored by the library atomic-server is using to configure these https://docs.rs/directories/latest/directories/struct.ProjectDirs.html#method.cache_dir, which is why we had to resort to adding `XDG_CACHE_HOME`.

Should we leave out these `*Directory` options then that won't actually impact the application? seems like they could be misleading

Reply:
The test fails without leaving `CacheDirectory` in as well, and I honestly can't explain it.